src/Data/Fixed.purs

```diff
@@ -271,10 +306,134 @@ approxDiv a b = Fixed (x * n / y)
     y = numerator b
     n = BigInt.fromInt (denominator a)
 
+-- | Parse a fixed-precision number from a string. Any decimal digits which are
+-- | not representable in the specified precision will be ignored.
+-- |
+-- | ```
+-- | > fromString "123.456" :: Maybe (Fixed P1000)
+-- | (Just (fromString "123.456" :: P1000))
+-- | ```
+-- |
+-- | Where possible, this function should be preferred over `fromNumber`, since
+-- | it is exact (whereas `fromNumber` can only provide an approximation for
+-- | larger inputs).
+-- |
+-- | ```
+-- | > fromString "9007199254740992.5" :: Maybe (Fixed P10)
+-- | (Just (fromString "9007199254740992.5" :: P10))
+-- |
+-- | > fromNumber 9007199254740992.5 :: Maybe (Fixed P10)
+-- | (Just (fromString "9007199254740992.0" :: P10))
+-- | ```
+-- |
+fromString
+  :: forall precision
+   . KnownPrecision precision
+  => String
+  -> Maybe (Fixed precision)
+fromString str =
+  let
+    numDigits = reflectPrecisionDecimalPlaces (PProxy :: PProxy precision)
+    denom = BigInt.fromInt (reflectPrecision (PProxy :: PProxy precision))
+
+    isDigit = between '0' '9'
+    wholeDigits = StringCU.countPrefix isDigit str
+    { before, after } = StringCU.splitAt wholeDigits str
+
+  in do
+    wholePart <-
+      BigInt.fromString before
+    fractionPart <-
+      case after of
+        "" -> pure zero
+        "." -> pure zero
+        _ -> do
+          guard (StringCU.charAt 0 after == Just '.')
+          let raw = StringCU.drop 1 after
+          BigInt.fromString (rightJustify numDigits '0' raw)
+    pure (Fixed (wholePart * denom + fractionPart))
+
+-- | Represent a `Fixed` value as a string, with the given number of decimal
+-- | places.
+-- |
+-- | ```
+-- | > map (toStringWithPrecision 2) (fromString "1234.567" :: Maybe (Fixed P1000))
+-- | (Just "1234.56")
+-- | ```
+-- |
+-- | If more decimal places are asked for than the type can provide, the extra
+-- | decimal places will be provided as zeroes.
+-- |
+-- | ```
+-- | > map (toStringWithPrecision 3) (fromString "1234.5" :: Maybe (Fixed P10))
+-- | (Just "1234.500")
+-- | ```
+toStringWithPrecision
+  :: forall precision
+   . KnownPrecision precision
+  => Int
+  -> Fixed precision
+  -> String
+toStringWithPrecision requestedDigits fixed@(Fixed n) =
+  let
+    denom = BigInt.fromInt (reflectPrecision (PProxy :: PProxy precision))
```

Comment:
could use `denominator` here, I think.

Reply:
Sounds good - I'll merge this and make that change in #12 so as to avoid conflicts?